wallets/index.js

```diff
@@ -1,452 +1,237 @@
+  }, [wallets, setWallets, me?.id])
 
-  const save = useCallback(async (newConfig) => {
-    await saveConfig(newConfig, { logger })
-  }, [saveConfig, me, logger])
-
-  // delete is a reserved keyword
-  const delete_ = useCallback(async (options) => {
-    try {
-      await clearConfig({ logger, ...options })
-    } catch (err) {
-      const message = err.message || err.toString?.()
-      logger.error(message)
-      throw err
-    }
-  }, [clearConfig, logger, disablePayments])
-
-  const deleteLogs_ = useCallback(async (options) => {
-    // first argument is to override the wallet
-    return await deleteLogs(options)
-  }, [deleteLogs])
-
-  if (!wallet) return null
+  useEffect(() => {
+    loadWallets()
+  }, [loadWallets])
 
-  // Assign everything to wallet object so every function that is passed this wallet object in this
-  // `useWallet` hook has access to all others via the reference to it.
-  // Essentially, you can now use functions like `enablePayments` _inside_ of functions that are
-  // called by `useWallet` even before enablePayments is defined and not only in functions
-  // that use the return value of `useWallet`.
-  wallet.isConfigured = _isConfigured
-  wallet.enablePayments = enablePayments
-  wallet.disablePayments = disablePayments
-  wallet.canSend = !!wallet.sendPayment
-  wallet.canReceive = !!wallet.createInvoice
-  wallet.config = config
-  wallet.save = save
-  wallet.delete = delete_
-  wallet.deleteLogs = deleteLogs_
-  wallet.setPriority = setPriority
-  wallet.hasConfig = hasConfig
-  wallet.status = status
-  wallet.enabled = enabled
-  wallet.priority = priority
-  wallet.logger = logger
-
-  // can't assign sendPayment to wallet object because it already exists
-  // as an imported function and thus can't be overwritten
-  return { ...wallet, sendPayment }
-}
-
-function extractConfig (fields, config, client) {
-  return Object.entries(config).reduce((acc, [key, value]) => {
-    const field = fields.find(({ name }) => name === key)
-
-    // filter server config which isn't specified as wallet fields
-    // (we allow autowithdraw members to pass validation)
-    if (client && key === 'id') return acc
-
-    // field might not exist because config.enabled doesn't map to a wallet field
-    if (!field || (client ? isClientField(field) : isServerField(field))) {
-      return {
-        ...acc,
-        [key]: value
-      }
-    } else {
-      return acc
-    }
-  }, {})
+  return { wallets, reloadLocalWallets: loadWallets, removeLocalWallets: removeWallets }
 }
 
-export function isServerField (f) {
-  return f.serverOnly || !f.clientOnly
-}
-
-export function isClientField (f) {
-  return f.clientOnly || !f.serverOnly
-}
-
-function extractClientConfig (fields, config) {
-  return extractConfig(fields, config, true)
-}
-
-function extractServerConfig (fields, config) {
-  return extractConfig(fields, config, false)
-}
+const walletDefsOnly = walletDefs.map(w => ({ def: w, config: {} }))
```

Comment:
We pull wallet definitions into an array so they can be treated like localWallets and serverWallets ... which is primarily useful for displaying them in the wallet grid.